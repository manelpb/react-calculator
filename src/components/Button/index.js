import styled from 'styled-components';

const H1 = styled.button`
  width: ${props => `${props.width}px`};
  height: ${props => `${props.height}px`};
  background-color: ${props => props.backgroundColor || 'transparent'};
  text-align: center;

  &:disabled {
    opacity: 0.5;
  }
`;

export default H1;