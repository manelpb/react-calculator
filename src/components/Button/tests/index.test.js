import React from 'react';
import Button from '../';
import { shallow, mount } from 'enzyme';
import 'jest-styled-components'
import styled from 'styled-components'

it('should render a html button', () => {
    const render = shallow(<Button />);
    expect(render.find('button').length).toEqual(1);
})

it('should render a html button with style', () => {
    const render = mount(<Button width={10} height={10} backgroundColor='red' />);

    expect(render).toHaveStyleRule('width', '10px');
    expect(render).toHaveStyleRule('height', '10px');
    expect(render).toHaveStyleRule('background-color', 'red');
})