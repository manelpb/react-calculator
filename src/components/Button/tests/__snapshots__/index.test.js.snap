// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a html button 1`] = `
.c0 {
  width: undefinedpx;
  height: undefinedpx;
  background-color: transparent;
  text-align: center;
}

.c0:disabled {
  opacity: 0.5;
}

<button
  className="c0"
/>
`;
